### PR TITLE
feat: add formik components, hooks and dependencies

### PR DESCRIPTION
**Pull request by chrisllontop**
Migrated all Formik components and their dependencies, as well as the hooks.

The latest versions of the following libraries were installed, updating their types in TypeScript:

- @popperjs/core
- date-fns
- focus-trap-react
- formik
- react-datepicker
- react-popper
- react-select
- react-select-event

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -113,7 +116,16 @@
   },
   "dependencies": {
     "@hyphen/hyphen-design-tokens": "^2.0.2",
+    "@popperjs/core": "^2.11.8",
     "classnames": "^2.5.1",
-    "react-select": "^5.8.0"
+    "cleave.js": "^1.6.0",
```
Comment (osbornm) [inline package.json:121]: This was causing problems with dependencies on next correct? We (@nathanyoung and I) said that we didn't need to worry about masking for now so we just should take out the masking that depended on this package.

**Comment by osbornm** [comment]
@nathanyoung Can you also take a look at this and see if you want anything changed?

File: package.json (hunk 1 above)
Comment (chrisllontop) [reply package.json:121]: Sure, I’ll remove it asap.

File: src/components/TextareaInput/TextareaInput.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+};
+
+export const HelpText = () => {
+  const [value, setValue] = useState('');
+  return (
+    <TextareaInput
+      id="helpText"
+      value={value}
+      label="Design Notes"
+      helpText="Add any design requirements or aesthetic choices that the solar system designer should know"
```
Comment (osbornm) [inline src/components/TextareaInput/TextareaInput.stories.tsx:79]:
```suggestion
      helpText="Add any additional comments that should be brought to the employees' attention"
```

File: src/components/CheckboxInput/CheckboxInput.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        label="This label is custom"
+        hideLabel
+        onChange={(event) => setValue(event.target.checked)}
+        isChecked={value}
+      />
+      <FormLabel inputId="customLabel">
+        <Box display="block" fontSize="xl" color="secondary-500">
+          This is custom, look at this{' '}
+          <a
+            href="https://palmetto.com"
```
Comment (osbornm) [inline src/components/CheckboxInput/CheckboxInput.stories.tsx:232]:
```suggestion
            href="https://hyphen.ai"
```

File: src/components/Table/Table.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,966 @@
...
+      },
+      { heading: 'phone', dataKey: 'phone', width: 125 },
+      { heading: 'plan', dataKey: 'plan', width: 125 },
+      { heading: 'status', dataKey: 'status', width: 125 },
+    ];
+    const tableData = [
+      {
+        status: 'lead captured',
+        name: 'John Smith',
+        email: 'john@palmetto.com',
```
Comment (osbornm) [inline src/components/Table/Table.stories.tsx:127]:
```suggestion
        email: 'john@hyphen.ai',
```
change all of these please

File: src/hooks/useBreakpoint/useBreakpoint.mdx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';
+import { ResponsiveProvider } from '../../components/ResponsiveProvider/ResponsiveProvider';
+import { useBreakpoint } from './useBreakpoint';
+import * as Stories from './useBreakpoint.stories';
+
+<Meta of={Stories} />
+
+# useBreakpoint
+
+The `useBreakpoint` returns useful states based on the window innerWidth and palmetto-component [breakpoints]().
```
Comment (osbornm) [inline src/hooks/useBreakpoint/useBreakpoint.mdx:12]:
```suggestion
The `useBreakpoint` returns useful states based on the window innerWidth and hyphen-component [breakpoints]().
```

File: src/hooks/useWindowSize/useWindowSize.mdx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { Meta, Canvas } from '@storybook/addon-docs';
+import { useWindowSize } from './useWindowSize';
+import * as Stories from './useWindowSize.stories';
+
+<Meta of={Stories} />
+
+# useWindowSize
+
+The `useWindowSize` returns useful states based on the window innerWidth and palmetto-component [breakpoints]().
```
Comment (osbornm) [inline src/hooks/useWindowSize/useWindowSize.mdx:9]:
```suggestion
The `useWindowSize` returns useful states based on the window innerWidth and hyphen-component [breakpoints]().
```

File: src/components/Formik/Formik.mdx [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Meta, Canvas } from '@storybook/addon-docs';
+import * as Stories from './Formik.stories';
+
+<Meta of={Stories} />
+
+# Formik Form
+
+Formik Form demonstrates a pattern for integrating the input elements found in Palmetto Components with [Formik](https://formik.org/). The form is fully functional and even mock submits. Manipulate the inputs in the form and watch the state object below it update with the new values in real-time. Attempt to submit the form without filling out required inputs, and observe the error states for these inputs.
```
Comment (osbornm) [inline src/components/Formik/Formik.mdx:8]:
```suggestion
Formik Form demonstrates a pattern for integrating the input elements found in Hyphen Components with [Formik](https://formik.org/). The form is fully functional and even mock submits. Manipulate the inputs in the form and watch the state object below it update with the new values in real-time. Attempt to submit the form without filling out required inputs, and observe the error states for these inputs.
```

File: src/components/Details/Details.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+          We are on a mission to stop climate change before we reach an
+          irreversible tipping point. Yes, this requires large shifts by entire
+          nations and entire populations. It requires urgency and change on a
+          massive scale. But technological advancements have put it within
+          reach.
+        </Box>
+      </Box>
+    </Details>
+  );
+};
```
Comment (osbornm) [inline src/components/Details/Details.stories.tsx:39]: Make something up that fits with Hyphen better.

**Review by nathanyoung — APPROVED** [review]
Looks good to me after comments are addressed. We can always modify these going forward.

**Pull request merged**